### DISCUSSION TITLE
My website deployed on Vercel keeps giving me the "429: TOO_MANY_REQUESTS" errors. It hasn't been launched yet, so it can't be having too much traffic. What could be going wrong, how can I debug this?

### DISCUSSION THREAD

**Discussion by lumenwrites** [vercel/next.js#31934] [Help] [answered]
Occassionally (maybe about half the time) when I load a page on the [website](https://nexy.io/) I'm working on, I'm getting an error that looks like [this](https://i.imgur.com/KEqtSQq.png).

> 429: TOO_MANY_REQUESTS  
> Code: INTERNAL_FUNCTION_RATE_LIMIT  
> ID: lhr1::258d8-1638206479250-0a01c8648601

My website hasn't been launched yet, almost nobody visits it but me, so it can't be having too much traffic yet.

The page I'm loading has a `getServerSideProps()` function that does only one thing - uses prisma to fetch posts from my database, which are sent to my component to be rendered.

I can't imagine what could be causing too many requests.

My vercel usage stats look like [this](https://i.imgur.com/pmWaMfR.png).

What am I doing wrong? What could be causing this? How can I debug this?

[Here's a gist](https://gist.github.com/lumenwrites/83d8fcac2623de6901bc6e730db36b09) where I have included all the relevant code. Any chance you could take a look at it and let me know if you have any ideas on what could be happening?

**Comment by Pieparker** [comment, marked as answer]
Hey there,

Please see this guide for more information: https://vercel.com/guides/what-should-i-do-if-i-receive-a-429-error-on-vercel

**Comment by tpetersons** [comment]
First thing that I would do is comment out prisma transaction and see if that is what is causing this issue.
To me it looks the only place in code that can cause some loops or timeouts etc.
Just try to return sample post and constant as count, remove the transaction and see if you get the same error.
If not, refer to Prisma docs and find what's up with the transaction and what options you have.

e.g. there are maxWait and timeout settings which might exceed Vercel`s
https://www.prisma.io/docs/concepts/components/prisma-client/transactions#example
